crates/turbo-tasks-fs/src/glob.rs

```diff
@@ -267,12 +269,9 @@ impl<'a> Iterator for GlobPartMatchesIterator<'a> {
         match self.part {
             GlobPart::AnyDirectories => {
                 if self.cursor.cur_cursor() == 0 {
-                    let end = self.cursor.next_boundary(self.path, 0);
-                    match end {
-                        Ok(Some(_)) => {}
-                        Ok(None) => return None,
-                        Err(..) => return None,
-                    }
+                    let Ok(Some(_)) = self.cursor.next_boundary(self.path, 0) else {
+                        return None;
+                    };
```

Comment:
This is just shorter and cleaner

Reply:
yes I'm always forgetting there's let..else syntax